src/Model/batchprocessing/batchprocessingMachineLearning/MachineLearningTrainingStage.py

```diff
@@ -0,0 +1,408 @@
+import pandas as pd
+import numpy as np
+
+# scroing
+from sklearn.metrics import (confusion_matrix, accuracy_score,
+                             f1_score, make_scorer)
+
+from imblearn.metrics import geometric_mean_score
+
+from sklearn.model_selection import GridSearchCV  # Fine Tune Model
+
+# classifiers Models
+from sklearn.ensemble import RandomForestClassifier
+from sklearn.neural_network import MLPClassifier
+
+# regression Models
+from sklearn.ensemble import RandomForestRegressor  # Random Forest
+from sklearn.neural_network import MLPRegressor  # MLP Regressor
+
+import logging
+import os
+import joblib
+
+
+class MlModeling():
+    def __init__(self,
+                 train_feature,
+                 test_feature,
+                 train_label,
+                 test_label,
+                 target,
+                 type_model,
+                 tunning=False,
+                 permission=None):
+        self.train_feature = train_feature
```

Comment:
do these have any definitions?
it looks like test_label is a list, otherwise you wouldn't be taking the mean of the difference between it and some other list on an element by element basis, squaring each of those, taking the mean, and then the square root.
Odds are someone who is already highly knowledgeable about machine learning will understand what all of these things mean.  Usually the people who take over existing code don't actually know that much about the domain (that's what the domain experts are for), and you want to give a programmer enough of an understanding of the variable types and how they will be used in the code so that if they have to go fix bugs, they'll at least know that much.

Reply:
It does.
I wrote it.